src/catalog/catalog_entry/schema_catalog_entry.cpp

```diff
@@ -140,9 +140,34 @@ void SchemaCatalogEntry::DropEntry(ClientContext &context, DropInfo *info) {
 }
 
 void SchemaCatalogEntry::AlterTable(ClientContext &context, AlterTableInfo *info) {
-	if (!tables.AlterEntry(context, info->table, info)) {
-		throw CatalogException("Table with name \"%s\" does not exist!", info->table.c_str());
-	}
+    switch (info->alter_table_type)	{
+        case AlterTableType::RENAME_TABLE: {
+            auto &transaction = Transaction::GetTransaction(context);
+            auto entry = tables.GetEntry(transaction, info->table);
+            if(entry->type == CatalogType::SCHEMA) {
```

Comment:
Shouldn't this be `entry->type != CatalogType::TABLE`?

Reply:
Yeah, for sure. I don't know where my head was at.
I will fix it.